features/step_definitions/administration_steps.rb

```diff
@@ -0,0 +1,115 @@
+Given(/^I have an admin account$/) do
+  FactoryGirl.create(:admin, email: "admin@example.com", password: "password")
+end
+
+Given(/^I do not have an admin account$/) do
+  Admin.delete_all
+end
+
+Given(/^I am logged into the admin panel$/) do
+  FactoryGirl.create(:admin, email: "admin@example.com", password: "password")
+  visit "/admin/sign_in"
+  fill_in "Email", with: "admin@example.com"
+  fill_in "Password", with: "password"
+  click_button "Log in"
+end
+
+Given(/^I am logged into the site$/) do
+  expect(page).to have_content("Signed in successfully")
+end
+
+Given(/^there is a book named "([^"]*)"$/) do |book_title|
+  FactoryGirl.create(:book, title: book_title )
+end
+
+Given(/^there is a book named "([^"]*)" valued at "([^"]*)"$/) do |book_title, price|
+  FactoryGirl.create(:book, title: book_title, price: price )
+end
+
+When(/^I click "([^"]*)" for the book "([^"]*)"$/) do |link_text, book_title|
+  find("div.row", text: book_title).click_link(link_text)
+end
+
+When(/^I visit the admin books url$/) do
+  visit "/admin/books"
+end
+
+When(/^I click "([^"]*)"$/) do |button_text|
+  click_button button_text
+end
+
+When(/^I enter the title "([^"]*)"$/) do |book_title|
```

Comment:
Nothing to change here and this is the right thing for this project. But as something else for you to know about, take a look at declarative vs imperative styles in cucumber testing (I think we actually talked about this a while ago). You can see Derek and I talking about it here: https://github.com/smashingboxes/guides/pull/3#discussion_r50276435, and you should also read this blog post http://benmabey.com/2008/05/19/imperative-vs-declarative-scenarios-in-user-stories.html that Derek also links to in the PR.